mongoengine/connection.py

```diff
@@ -103,7 +103,13 @@ def _get_connection_settings(
         if entity.startswith("mongomock://"):
             conn_settings["is_mock"] = True
             # `mongomock://` is not a valid url prefix and must be replaced by `mongodb://`
-            resolved_hosts.append(entity.replace("mongomock://", "mongodb://", 1))
+            new_entity = entity.replace("mongomock://", "mongodb://", 1)
+            resolved_hosts.append(new_entity)
+
+            uri_dict = uri_parser.parse_uri(new_entity)
+
+            if uri_dict.get("database"):
```

Comment:
it's very minor but extract `database_name` in a variable to avoid repetition of the .get
its slightly longer but it's more readable

Reply:
I didn't do it for consistency, https://github.com/MongoEngine/mongoengine/blob/290d9df3eb6491f22a8feea0f2e9e909bc94b621/mongoengine/connection.py#L120 is using the same pattern.

I don't mind changing it but then we need to change it in that line too.

Do i still change it?

Reply:
sorry for the delayed reply, yes please change that in the 2 places